harness/propertyHelper.js

```diff
@@ -1,13 +1,17 @@
 
 function isConfigurable(obj, name) {
+    var descriptor = Object.getOwnPropertyDescriptor(obj, name);
     try {
         delete obj[name];
     } catch (e) {
         if (!(e instanceof TypeError)) {
             $ERROR("Expected TypeError, got " + e);
         }
     }
-    return !Object.prototype.hasOwnProperty.call(obj, name);
+    var result = !Object.prototype.hasOwnProperty.call(obj, name);
+    // Restore property (no-op if not deleted)
+    Object.defineProperty(obj, name, descriptor);
```

Comment:
> Well, would you object to adding a function which just does the test and doesn't do the delete?

I agree with you this would be the proper approach. I really want to have some that only checks for `Object.getOwnPropertyDescriptor(obj, 'prop').configurable;`, but that's when we got the history of finding false positives from runtimes.

I have plans to create an easier to consume harness api for properties, where I would verify it like:

```js
verifyProperty(obj, 'prop', {
  configurable: false,
  enumerable: true,
  writable: false
});
```

So the tests won't need to worry about the order. e.g. `verifyWritable` being called after `verifyConfigurable`.

Using this same function, we can extend it with options, let's say:

```js
verifyProperty(obj, 'prop', {
  configurable: false,
  enumerable: true,
  writable: false
}, {
  descriptorOnly: true // avoid changing the actual values or deleting properties
});
```

or even: 

```js
verifyProperty(obj, 'prop', {
  configurable: false,
  enumerable: true,
  writable: false,
  __descriptorOnly: true
});
```

The function signature is only a bikeshed, but a new PR for it might be a good starting point to discuss the options.

Does it work for you?